src/index.ts

```diff
@@ -36,17 +36,43 @@ const listen =
       type: "LISTEN",
     });
 
+interface Grammar {
+  [index: string]: {
+    entities: {
+      [index: string]: string;
+        };
+      };
+    }
+
+    const grammar = {
```

Comment:
how does this scale? It seems more like an interpretation of one-shot request "I want to travel from stockholm to helsinki on wednesday". What would happen if there are more days and origins/destinations? Please add a couple of other options. 

Please, also let me know what I should say to the system. I said "travel" and it was interpreted as destination = undefined. 
 
Nevertheless, it seems you are on the right track with the state machine.